### PR TITLE
Add streaming dependencies to 'umd_comms' + RTSP server HTTP API feature

### PR DESCRIPTION
**Pull request by amoramar**
NOTE: this PR was branched from from `feat/rtsp-http-api-integration` (open PR here: https://github.com/umdlife/darknet_ros/pull/2). Thus, this is a dual-purpose PR:
- Bump `umd_streaming` to latest commit, in which the streaming dependencies from `umd_libs` were migrated to a new package, `umd_comms`
- Implementation of the RTSP server HTTP API feature, wtih two main points:
    - Dynamic generation of RTSP URLs, using the HTTP API of the RTSP server (located at `RTSP_SERVER_IP` IP).
    - Developer tool to set custom **writer** GStreamer pipelines.

---
 
## Basic Info
 
| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | |
| Primary OS tested on | Ubuntu 20.04 LTS |
| Robotic platform tested on |  |
 
---
## How to run this PR feature
TBT alongisde https://github.com/umdlife/umd_streaming/pull/28

 
## Description of contribution in a few bullet points
 Refer to https://github.com/umdlife/umd_streaming/pull/28

 
## Description of documentation updates required from your changes
 Refer to https://github.com/umdlife/umd_streaming/pull/28

 
---
 
## Future work that may be required in bullet points
 Refer to https://github.com/umdlife/umd_streaming/pull/28

### CODE REVIEW

File: src/YoloObjectDetector.cpp [hunk 1]
```diff
@@ -136,11 +154,30 @@ void YoloObjectDetector::on_image_callback(const cv::Mat& image)
     frameHeight_ = image.rows;
     if (!writer_configured_)
     {
-      std::string url_writer(this->get_namespace());
-      url_writer = "rtsp://127.0.0.1:8554"+url_writer;
-      std::cout << "url_writer: " << url_writer << std::endl;
-      // Configure RTSP Streamer
-      rtsp_streamer_.on_configure_writer(frameWidth_, frameHeight_, 30, 9000, url_writer);
+      // Check whether a GStreamer pipeline has been provided
+      std::string gstreamer_pipeline;
```
Comment (vicmassy) [inline src/YoloObjectDetector.cpp:158]: We also need to be able to configure the reader pipeline. We already discussed about it but I cannot see the changes.

**Review by vicmassy — CHANGES_REQUESTED** [review]

File: src/YoloObjectDetector.cpp (hunk 1 above)
Comment (amoramar) [reply src/YoloObjectDetector.cpp:158]: @vicmassy I left a comment on another `umd_darknet` PR; the main point is that `umd_rtsp` does not offer an `on_configure_reader` method that accepts a custom pipeline as an input. https://github.com/umdlife/darknet_ros/pull/2#discussion_r1021202706

If we want to do that, we would need to open a new PR on `umd_rtsp` to implement that, then come back to this point.

**Review by vicmassy — APPROVED** [review]
RTBM! Open an issue regarding extending the custom pipeline capability to the reader pipeline too.

**Comment by amoramar** [comment]
> RTBM! Open an issue regarding extending the custom pipeline capability to the reader pipeline too.

For reference: https://github.com/umdlife/umd_streaming/issues/29

**Pull request merged**